### PR TITLE
teensy: add west flash teensy_loader_cli integration

### PR DESCRIPTION
**Pull request by PetervdPerk-NXP**
Allows Teensy 4.0 and 4.1 board to be flashed using the `west flash` command.

### CODE REVIEW

**Comment by PetervdPerk-NXP** [comment]
I've got CI in an okayish state but I'm not sure how to resolve this pylint error. Could someone help me further?

File: scripts/west_commands/runners/teensy.py [hunk 1]
```diff
@@ -0,0 +1,79 @@
...
+            self.flash(**kwargs)
+
+    def flash(self, **kwargs):
+        if self.hex_name is not None and os.path.isfile(self.hex_name):
+            fname = self.hex_name
+        else:
+            raise ValueError(
+                'Cannot flash; no hex ({}) files found. '.format(self.hex_name))
+
+        cmd = ([self.teensy] +
```
Comment (danieldegrasse) [inline scripts/west_commands/runners/teensy.py:66]: I don't think using `-w` here with the `while True` loop here makes sense. My concern is that the loader will effectively freeze waiting for the device to appear. There are a few issues with this:
- if the image being flashed is large, the user may confuse the loader waiting with the image taking time to flash
- if the device is: not connected, not powered, or not in HalfKay bootloader mode there is no feedback to the user
- this same issue appears if the device is not accessible (IE if the udev rules don't give the user access to the teensy device)

Comment (PetervdPerk-NXP) [reply scripts/west_commands/runners/teensy.py:66]: Thanks for the suggestions, I've removed the loop and -w option.
On success output is
```
[1/1] cd /home/peter/cognipilot/ws/zephyr/build/zephyr && ...e=/home/peter/cognipilot/ws/zephyr/build/zephyr/zephyr.elf
-- west flash: using runner teensy
-- runners.teensy: Flashing file: /home/peter/cognipilot/ws/zephyr/build/zephyr/zephyr.hex
-- runners.teensy: Success
```
On failure you get this
```
[1/1] cd /home/peter/cognipilot/ws/zephyr/build/zephyr && ...e=/home/peter/cognipilot/ws/zephyr/build/zephyr/zephyr.elf
-- west flash: using runner teensy
-- runners.teensy: Flashing file: /home/peter/cognipilot/ws/zephyr/build/zephyr/zephyr.hex
Unable to open device (hint: try -w option)

FATAL ERROR: command exited with status 1: teensy_loader_cli -v --mcu TEENSY41 /home/peter/cognipilot/ws/zephyr/build/zephyr/zephyr.hex
```

File: scripts/west_commands/runners/teensy.py [hunk 2]
```diff
@@ -0,0 +1,77 @@
...
+        self.hex_name = cfg.hex_file
+
+    @classmethod
+    def name(cls):
+        return 'teensy'
+
+    @classmethod
+    def capabilities(cls):
+        return RunnerCaps(commands={'flash'},
+                          dev_id=True, flash_addr=True, erase=True,
```
Comment (danieldegrasse) [inline scripts/west_commands/runners/teensy.py:36]: It doesn't look like this runner supports these capabilities, unless I'm missing something? If not, we can drop this line- all these capabilities default to False. Same thing applies to `tool_opt`

File: scripts/west_commands/runners/teensy.py [hunk 3]
```diff
@@ -0,0 +1,77 @@
...
+                            help='Teensy muc target')
+        parser.add_argument('--teensy', default='teensy_loader_cli',
+                            help='path to teensy cli tool, default is teensy_loader_cli')
+
+    @classmethod
+    def do_create(cls, cfg, args):
+        ret = TeensyBinaryRunner(
+            cfg, args.mcu,
+            teensy=args.teensy,
+            erase=args.erase)
```
Comment (danieldegrasse) [inline scripts/west_commands/runners/teensy.py:51]: Is this used?

**Comment by PetervdPerk-NXP** [comment]
@danieldegrasse thank you for the suggestions I've cleaned-up the python code could you take a look again?

**Review by bperseghetti — DISMISSED** [review]

File: boards/pjrc/teensy4/board.cmake [hunk 4]
```diff
@@ -0,0 +1,11 @@
...
+
+board_set_flasher_ifnset(teensy)
+
+if(CONFIG_BOARD_TEENSY40)
+board_runner_args(teensy "--mcu=TEENSY40")
+else()
+board_runner_args(teensy "--mcu=TEENSY41")
+endif()
+
+include(${ZEPHYR_BASE}/boards/common/teensy.board.cmake)
```
Comment (danieldegrasse) [inline boards/pjrc/teensy4/board.cmake:11]: Nit- missing newline at end of file

File: scripts/west_commands/tests/test_imports.py [hunk 5]
```diff
@@ -43,6 +43,7 @@ def test_runner_imports():
                     'spi_burn',
                     'stm32cubeprogrammer',
                     'stm32flash',
+                    'teensy',
```
Comment (danieldegrasse) [inline scripts/west_commands/tests/test_imports.py:46]: @nashif it seems that the addition of `teensy` to this file and `__init__.py` is flagging the pylint duplicate code check, causing CI to fail. Should we just ignore the CI failure in this case? Or is there a fix we could apply to the pylint check?

Comment (PetervdPerk-NXP) [reply scripts/west_commands/tests/test_imports.py]: Splitting it up into 2 commits seems to solve it.

Comment (danieldegrasse) [reply scripts/west_commands/tests/test_imports.py]: Perfect, wasn't sure if CI was running per commit or looking at the entire PR diff

**Review by danieldegrasse — APPROVED** [review]

**Review by bperseghetti — APPROVED** [review]

**Comment by mmahadevan108** [comment]
@mbolivar-ampere, can you help take a look at this PR.

**Comment by dleach02** [comment]
@mbolivar-ampere, moved this over to me as assignee as it is mostly a platform with an NXP part on it. If you object then please adjust back.

**Review by dleach02 — APPROVED** [review]

**Pull request merged**